src/pgen/default_pgen.cpp

```diff
@@ -131,6 +131,18 @@ void Mesh::UserMeshWorkBeforeOutputDefault(Mesh *, ParameterInput *, SimTime con
   return;
 }
 
+//========================================================================================
+//! \fn void Mesh::UserMeshWorkBeforeRestartOutputDefault(Mesh *pmb, ParameterInput *pin,
+//! SimTime &t)
+//  \brief Function called before generating output files
+//========================================================================================
+
+void Mesh::UserMeshWorkBeforeRestartOutputDefault(Mesh *, ParameterInput *,
+                                                  SimTime const &) {
+  // do nothing
+  return;
+}
```

Comment:
I don't think these weakly linked defaults are needed or used anymore are they? We can just put the function pointer in `application_input`?

Would it be desirable also to have a `StateDescriptor` hook for per-package work before restart output?

Reply:
You're right, these aren't needed, and are removed.

I agree that a `StateDescriptor` hook in addition is desirable. Just added.